### ISSUE TITLE
Q.real(x) does not lead to Q.finite(x) being true

### ISSUE THREAD

**Issue by oscargus** [sympy/sympy#17113]
It may be that I have no idea about how the assumption system is supposed to work, but I was under the impression that `ask(Q.finite(x), Q.real(x))` should evaluate to `True`.

**Comment by jksuom** [comment]
It looks like the implication `Q.finite(x) | ~Q.real(x)` is missing from known facts: https://github.com/sympy/sympy/blob/26aa686a20234f1f511c0f558d27665d0f95a21c/sympy/assumptions/ask_generated.py#L28-L31

**Comment by oscargus** [comment]
Yes, that was my conclusion as well. That is `Implies(Q.real, Q.finite)`, right?

**Comment by oscarbenjamin** [comment]
I think I thought this was already the case in the new assumptions. Yes that implication should be added and then ask_generated should be rebuilt.

**Comment by TiloRC** [comment]
This seems to be correctly handled now. The following returns True.
```python
from sympy import ask, Q
from sympy.abc import x
ask(Q.finite(x), Q.real(x))
```
